roles/vivumlab_wireguard/tasks/main.yml

```
@@ -1,19 +1,37 @@
 ---
-- name: Add the WireGuard PPA repository
+- name: Add the WireGuard PPA repository (Debian)
   apt_repository:
-    repo: ppa:wireguard/wireguard
+    repo: deb http://deb.debian.org/debian buster-backports main
     state: present
-  register: result
-  until: result is succeeded
-  retries: 10
-  delay: 3
+    update_cache: yes
+  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "10")
 
-- name: Install WireGuard
+- name: Add the WireGuard PPA repository (Debian)
+  apt_repository:
+    repo: deb http://deb.debian.org/debian/ unstable main
+    state: present
+    update_cache: yes
+  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] !== "10" and ansible_facts[distribution_major_version] !== "11")
```

```suggestion
  when: (ansible_facts['distribution'] == "Debian") and (ansible_facts[distribution_major_version] != "10" or ansible_facts[distribution_major_version] != "11")
```
Did you test this? Unequal usually looks like this `!=`.